spec/require_spec.rb

```diff
@@ -1,26 +1,73 @@
 require File.dirname(__FILE__) + '/spec_helper.rb'
 require File.dirname(__FILE__) + '/require_shared.rb'
 
+shared_examples_for 'strict mode' do |method, fixture_method|
+  context 'strict error mode' do
+    context 'using global option' do
+      def with_strict_mode
+        RequireAll.strict_mode = true
+        yield
+      ensure
+        RequireAll.strict_mode = false
+      end
+
+      it "can require resolvable dependencies when there are no errors raised" do
+        with_strict_mode do
+          fixture = send(fixture_method, 'resolvable_with_strict_mode/*.rb')
+          send(method, fixture)
+          is_expected.to be_loaded("A", "B", "C", "D")
+        end
+      end
+
+      it "raises errors when they occur" do
+        with_strict_mode do
+          fixture = send(fixture_method, 'resolvable/*.rb')
+          expect do
+            send(method, fixture)
+          end.to raise_error(NameError)
+        end
+      end
+    end
+
+    context 'using options on method call' do
+      it "can require resolvable dependencies when there are no errors raised" do
+        fixture = send(fixture_method, 'resolvable_with_strict_mode/*.rb')
+        send(method, fixture, :strict_mode => true)
```

Comment:
Can/should we use new-style symbol key syntax (`strict_mode: true`)

Reply:
Ok, just decided that current [stable versions of Ruby](https://www.ruby-lang.org/en/downloads/) should be supported in which case using "new-style" is ok. Updated [CI configuration](https://github.com/jarmo/require_all/blob/master/.travis.yml) accordingly.

Still, I'd introduce that change globally in a separate PR.

Reply:
@jarmo done! https://github.com/jarmo/require_all/pull/22